clients/venice-client/src/main/java/com/linkedin/venice/fastclient/RetriableAvroGenericStoreClient.java

```diff
@@ -315,6 +345,7 @@ private <R extends MultiKeyRequestContext<K, V>, RESPONSE> void retryStreamingMu
             savedException,
             pendingKeysFuture,
             scheduledRetryTask));
+    longTailRetryManager.recordRequest();
```

Comment:
Currently retry budget is at the request level, perhaps we should maintain it at the key level and use KPS then?

Reply:
I think request level is fine, and KPS level will be problematic if not considering request type.
For example, avg key count for multi-get request is 1000 and if the key based retry budget is being utlized by single-get request, it will allow a very high number of single get requests, which will be very bad.